src/python/lib/ensembl/compara/orthtools/orthofinder.py

```diff
@@ -0,0 +1,92 @@
+"""Methods for preparing input files and running OrthoFinder.
+
+This module provides::
+
+    :func:`prepare_input_orthofinder()` which copies input fasta files for OrthoFinder
+            into a specified working directory
+
+    :func:`run_orthofinder()` which runs OrthoFinder
+
+Typical usage example::
+
+    from ensembl.compara.orthtools import prepare_input_orthofinder
+    prepare_input_orthofinder('path/to/source_dir', 'path/to/target_dir')
+
+    from ensembl.compara.orthtools import run_orthofinder
+    run_orthofinder('path/to/input/fasta/files')
+
+    from ensembl.compara.orthtools import run_orthofinder
+    run_orthofinder('path/to/input/fasta/files', 16, 4)
+
+"""
+
+__all__ = ['prepare_input_orthofinder', 'run_orthofinder']
+
+import fnmatch
+import os
+from pathlib import Path
+import subprocess
+
+def prepare_input_orthofinder(source_dir: str, target_dir: str) -> None:
+    """Creates symlinks to input fasta files in `target_dir`.
+
+    Args:
+        source_dir: Path to the directory containing fasta files.
+        target_dir: Path to the directory where symlinks to fasta files will be created
+                for OrthoFinder to use.
+
+    Raises:
+        FileExistsError: If directory `target_dir` already exists.
+        FileNotFoundError: If directory `source_dir` does not exist or does not contain any fasta files.
+        subprocess.CalledProcessError: If creating symlinks fails for some other reason.
+
+    """
+    if not os.path.isdir(source_dir):
+        raise FileNotFoundError("Directory containing fasta files not found.")
+    if len(fnmatch.filter(os.listdir(source_dir), '*.fasta')) == 0:
+        raise FileNotFoundError("No fasta files found.")
+
+    # To ensure a new directory is used for running OrthoFinder:
+    os.mkdir(target_dir)
+
+    script = os.path.join(Path(__file__).parents[6], "scripts", "pipeline", "symlink_fasta.py")
+
+    subprocess.run([script, "-s", target_dir, "-d", source_dir], capture_output=True, check=True)
+
+
+def run_orthofinder(input_dir: str, number_of_threads: int = 32, number_of_orthofinder_threads: int = 8) \
+        -> None:
+    """Runs OrthoFinder.
+
+    Args:
+        input_dir: Path to the directory containing input fasta files.
+        number_of_threads: The number of parallel processes for the BLAST/DIAMOND searches and
+                tree inference steps.
+        number_of_orthofinder_threads: The number of parallel processes for other OrthoFinder steps
+                that have been parallelised.(If not specified at all, defaults to
+                min{16, 1/8 * `number_of_threads`}.)
+
+    Raises:
+        FileNotFoundError: If OrthoFinder executable (`orthofinder_exe`) cannot be found.
+        subprocess.CalledProcessError: If executing OrthoFinder command fails for some reason,
+                including `input_dir` not found.
+
+    """
+    orthofinder_exe = "/hps/software/users/ensembl/ensw/C8-MAR21-sandybridge/linuxbrew/bin/orthofinder"
```

Comment:
Would it be a good idea to concentrate harder code path on a shared/common python file?

Reply:
It would be easy if things changes...

Reply:
I think I will do something like that when I move this pipeline into NextFlow 👍

Reply:
Sorry, nosy here: and if it shares NextFlow config for params etc (as one would expect)- this is already existing in the repo :innocent: